packages/protocol-perf/src/index.ts

```diff
@@ -147,23 +189,48 @@ class DefaultPerfService implements Startable, PerfService {
 
     const stream = await connection.newStream([this.protocol], options)
 
+    let lastAmountOfBytesSent = BigInt(0)
+    let lastReportedTime = Date.now()
+    let totalBytesSent = BigInt(0)
+
+    const initialStartTime = Date.now()
+
     // Convert sendBytes to uint64 big endian buffer
     const view = new DataView(this.databuf)
     view.setBigInt64(0, recvBytes, false)
 
     log('sending %i bytes to %p', sendBytes, connection.remotePeer)
+
     try {
       await stream.sink((async function * () {
         // Send the number of bytes to receive
         yield uint8Buf.subarray(0, 8)
-        // Send the number of bytes to send
+
         while (sendBytes > 0n) {
           let toSend: bigint = writeBlockSize
           if (toSend > sendBytes) {
             toSend = sendBytes
           }
           sendBytes = sendBytes - toSend
           yield uint8Buf.subarray(0, Number(toSend))
+
+          if (Date.now() - lastReportedTime > 1000) {
+            const output: PerfOutput = {
+              type: 'intermediary',
+              timeSeconds: (Date.now() - lastReportedTime) / 1000,
+              uploadBytes: lastAmountOfBytesSent.toString(),
+              downloadBytes: BigInt(0).toString()
+            }
+
+            lastReportedTime = Date.now()
+            lastAmountOfBytesSent = BigInt(0)
+
+            // eslint-disable-next-line no-console
+            console.log(JSON.stringify(output))
```

Comment:
Same as https://github.com/libp2p/js-libp2p/pull/2067/files#r1373864207 - we should not be writing anything out to the console during a measured benchmark.

Reply:
Referencing comment above: https://github.com/libp2p/js-libp2p/pull/2067#discussion_r1377845934

Reply:
As @mxinden [mentioned](https://github.com/libp2p/js-libp2p/pull/2067#discussion_r1377848293) we are not benchmarking CPU performance but just the amount of bytes that are being uploaded vs downloaded so I am not sure if this is a worthwhile optimization.